ompi/communicator/comm.c

```diff
@@ -198,15 +207,45 @@ int ompi_comm_set_nb ( ompi_communicator_t **ncomm,
         }
 
         newcomm->c_flags |= OMPI_COMM_INTER;
+        newcomm->c_index_vec = malloc(remote_size * sizeof(int));
```

Comment:
This is one sneaky change, with drastic implications in terms of latency, scalability and memory usage. It need to be carefully assessed and discusses before being merged.

Reply:
yes agreed.   There's an issue to address this in a different way - https://github.com/hpc/ompi/issues/65